Userland/Services/WebContent/WebDriverConnection.cpp

```diff
@@ -2118,26 +2118,35 @@ Messages::WebDriverClient::PrintPageResponse WebDriverConnection::print_page(Jso
     return Web::WebDriver::Error::from_code(Web::WebDriver::ErrorCode::UnsupportedOperation, "Print not implemented"sv);
 }
 
-Messages::WebDriverClient::EnsureTopLevelBrowsingContextIsOpenResponse WebDriverConnection::ensure_top_level_browsing_context_is_open()
+// https://w3c.github.io/webdriver/#dfn-set-the-current-browsing-context
+void WebDriverConnection::set_current_browsing_context(Web::HTML::BrowsingContext& browsing_context)
 {
-    TRY(ensure_current_top_level_browsing_context_is_open());
-    return JsonValue {};
+    // 1. Set session's current browsing context to context.
+    m_current_browsing_context = browsing_context;
+
+    // 2. Set the session's current parent browsing context to the parent browsing context of context, if that context
+    //    exists, or null otherwise.
```

Comment:
Should this have an `else` to set to `nullptr` for the otherwise case?

Reply:
Yep!